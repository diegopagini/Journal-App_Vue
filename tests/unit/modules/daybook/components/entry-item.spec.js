/** @format */
import EntryItem from '@/modules/daybook/components/entry-item.vue';
import { shallowMount } from '@vue/test-utils';

import { journalState } from '../../../mock-data/test-journal-state';

describe('Entry item component', () => {
	let wrapper;
	const mockRouter = {
		push: jest.fn(),
	};

	beforeEach(() => {
		wrapper = shallowMount(EntryItem, {
			props: {
				entry: journalState.entries[0],
			},
			global: {
				mocks: {
					$router: mockRouter,
				},
			},
		});
	});

	it('should match with the snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('shoul redirect after click in the entry-container', () => {
		const entryContainer = wrapper.find('.entry-container');
		entryContainer.trigger('click');
		expect(mockRouter.push).toHaveBeenCalled();
		expect(mockRouter.push).toHaveBeenCalledWith({
			name: 'entry',
			params: {
				id: journalState.entries[0].id,
			},
		});
	});

	it('should have the computed properties', () => {
		expect(wrapper.vm.day).toBe(14);
		expect(wrapper.vm.month).toBe('Agosto');
		expect(wrapper.vm.yearDay).toBe('2023, Lunes');
	});
});
