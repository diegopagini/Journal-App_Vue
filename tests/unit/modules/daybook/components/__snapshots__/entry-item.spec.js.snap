// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry item component should match with the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">14</span><span class="mx-1 fs-5">Agosto</span><span class="mx-2 fw-light">2023, Lunes</span></div>
  <div class="entry-description">Es un d√≠a bonito!</div>
</div>
`;
